system/default.nix

```diff
@@ -73,12 +98,51 @@
     ];
   };
 
+  # Enable podman with DNS
+  virtualisation.podman = {
+    enable = true;
+  } // (
+    # dnsname allows containers to use ${name}.dns.podman to reach each other
+    # on the same host instead of using hard-coded IPs.
+    # NOTE: --net must be the same on the containers, and not eq "host"
+    # TODO: extend this with flannel ontop of wireguard for cross-node comms
+    if (lib.trivial.release == "22.11") then {
+      defaultNetwork.dnsname.enable = true;
+    } else {
+      defaultNetwork.settings = { dns_enabled = true; };
+    }
+  );
```

Comment:
This if as the one above can probably be removed now that the main stable release is 23.05.